### PR TITLE
require definition of EBML Header/Globals in EBML Schema

### PR DESCRIPTION
**Pull request by dericed**
related to https://github.com/Matroska-Org/ebml-specification/issues/128

### CODE REVIEW

File: ebml_schema_example.xml [hunk 1]
```diff
@@ -1,6 +1,31 @@
...
+ <element name="EBMLMaxSizeLength" path="1*1(\EBML\EBMLMaxSizeLength)"
+   id="0x42F3" minOccurs="1" maxOccurs="1" range="not 0" default="8"
+   type="uinteger"/>
+ <element name="DocType" path="1*1(\EBML\DocType)" id="0x4282"
+   minOccurs="1" maxOccurs="1" size=">0" type="string"/>
+ <element name="DocTypeVersion" path="1*1(\EBML\DocTypeVersion)"
+   id="0x4287" minOccurs="1" maxOccurs="1" default="1" type="uinteger"/>
+ <element name="DocTypeReadVersion" path="1*1(\EBML\DocTypeReadVersion)"
+   id="0x4285" minOccurs="1" maxOccurs="1" default="1" type="uinteger"/>
+ <element name="CRC-32" path="1*1(\EBML\EBMLReadVersion)" id="0xBF"
```
Comment (robUx4) [inline ebml_schema_example.xml]: I think the path for the CRC-32 is wrong

**Comment by dericed** [comment]
Oops, may be removing the line altogether since there's no Matroska constaint on EBML's CRC. Will ping once updated.

**Comment by dericed** [comment]
This PR is updated based upon @robUx4's comments at https://github.com/Matroska-Org/matroska-specification/pull/135#issuecomment-307703820. Now the documentation only mandates EBML Header elements are defined in an EBML Schema if there is an added constraint to the use of that element. Also only constraints to the range attribute are allowed. Thus the matroska EBML Schema can constrain the range of EBMLMaxSizeLength to 8, but doesn't need to document all the other Header Elements as well. I left out constraints to EBML's global elements since it doesn't really make sense to constrain them.

File: specification.markdown [hunk 2]
```diff
@@ -279,7 +279,9 @@ An `EBML Schema` is an XML Document that defines the properties, arrangement, an
...
+The `EBML Schema` MUST document all Elements of the `EBML Body`. The `EBML Schema` does not document `Global Elements` that are defined by this document (namely the `Void Element` and the `CRC-32 Element`).
+
+An `EBML Schema` MAY constrain the use of `EBML Header Elements` (see [EBML Header Elements](#ebml-header-elements)) by adding or constraining that Element's `range` attribute. For example, an `EBML Schema` MAY constrain the `EBMLMaxSizeLength` to a maximum value of `8` or MAY constain the `EBMLVersion` to only support a value of `1`. If an `EBML Schema` adopts the `EBML Header Element` as-is, then it is not REQUIRED to document that Element within the `EBML Schema`. If an `EBML Schema` constrains the range of an `EBML Header Element` then that `Element` MUST be documented within an `<element>` node of the `EBML Schema`. This document provides an example of an `EBML Schema`, see [EBML Schema Example](#ebml-schema-example).
```
Comment (mbunkus) [inline specification.markdown]: A missing comma before "then that `Element` MUST be documented…"

**Comment by dericed** [comment]
Updated per @mbunkus' comment at https://github.com/Matroska-Org/ebml-specification/pull/152#discussion_r125188430.

**Pull request merged**